types/src/reconfiguration_event_subscription.rs

```diff
@@ -0,0 +1,9 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+/// Trait that must be implemented by a subscription to reconfiguration events emitted
+/// by state sync
+
+pub trait ReconfigurationEventSubscription: Send + Sync {
+    fn publish(&self, payload: usize);
```

Comment:
the payload probably should be ContractEvent or some generic?

Reply:
Yes, this will be fleshed out once how exactly the reconfiguration event will be represented in the node. But I can make it `Vec<ContractEvent>` for now to make it more meaningful